gnocchi/indexer/__init__.py

```diff
@@ -69,6 +69,8 @@ def lastmodified(self):
         return self.revision_start.replace(microsecond=0,
                                            tzinfo=iso8601.iso8601.UTC)
 
+    __hash__ = object.__hash__
```

Comment:
uh, what's the point of this hash? is this overriding anything?

Reply:
On python3, when __eq__ is defined we must defined __hash__ otherwise hashing is rejected.

> If a class does not define an __eq__() method it should not define a __hash__() operation either; if it defines __eq__() but not __hash__(), its instances will not be usable as items in hashable collections

Since I use Resource in set(), I have to define __eq__ and __hash__.

Reply:
is this just a random hash though or is there some logic to how hash is generated? if it's random, i feel like we should be explicit on how the hash is generated ie. hash the same attributes as __eq__

Reply:
By default, it's just random. In my case I just want to identify if two references point to the same object. I don't want to consider equal two different objects describing the same resource.